event.c

```diff
@@ -1977,6 +1998,11 @@ event_base_loop(struct event_base *base, int flags)
 			evutil_timerclear(&tv);
 		}
 
+		/* Invoke check watchers after polling for events, and before processing them */
```

Comment:
This is definitely in the wrong place. It should be further down, after `evsel->dispatch(...)`